src/Console/Installer.php

```diff
@@ -156,7 +156,17 @@ public static function setFolderPermissions($dir, $io)
             }
         };
 
-        $worldWritable = bindec('0000000111');
+        $worldWritable = 007;
+        if ($worldWritable === 007) {
```

Comment:
Maybe I missunderstood something but `composer update` will update `composer.lock` whereas `composer install` will simply read `composer.lock` which is exactly what you want after deploying an application source update (frozen vendor dependencies might have changed, or not... fire the command and be save as composer.lock went into the git repository).